From a0a2012b203b5c1624266cda476f5c1a85c0c8d6 Mon Sep 17 00:00:00 2001
From: "sameer.mulla" <sameer.mulla@lge.com>
Date: Wed, 11 Apr 2018 11:03:20 +0530
Subject: [PATCH] Fix advertise time out when default is set to zero

:Release Notes:
Advertising timeout zero is ignored

:Detailed Notes:
Advertising default value for timeout parameter is zero
so zero timing out is ignored

:Testing Performed:
Builded and tested

:QA Notes:

:Issues Addressed:
[PLAT-51859] Implement method ble/startAdvertise

Change-Id: I60cb0c22991efa61a800b1483e232e42beff2da2
---
Upstream-Status: Pending

 src/advertising.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/advertising.c b/src/advertising.c
index 2c9a5a443..efd795712 100644
--- a/src/advertising.c
+++ b/src/advertising.c
@@ -569,13 +569,18 @@ static bool parse_local_name(DBusMessageIter *iter,
 	if (dbus_message_iter_get_arg_type(iter) != DBUS_TYPE_STRING)
 		return false;
 
+	dbus_message_iter_get_basic(iter, &name);
+
+	if (name && !strcmp(name, "")) {
+		DBG("It's default prop name ignore");
+		return true;
+	}
+
 	if (client->flags & MGMT_ADV_FLAG_LOCAL_NAME) {
 		error("Local name already included");
 		return false;
 	}
 
-	dbus_message_iter_get_basic(iter, &name);
-
 	free(client->name);
 
 	/* Treat empty string the same as omitting since there is no point on
